CoreFeatures/Database-postgresql-plugins/PluginPostgresConnectionPool/src/main/java/info/smart_tools/smartactors/database_postgresql_plugins/postgres_connection_pool_plugin/PostgresConnectionPoolPlugin.java

```diff
@@ -69,14 +69,25 @@ public void load() throws PluginException {
                                         return IOC.resolve(poolKey);
                                     } catch (ResolutionException re) {      // pool not found
                                         try {
-                                            IPool pool = new Pool(connectionOptions.getMaxConnections(), () -> {
-                                                try {
-                                                    return new PostgresConnection(connectionOptions);
-                                                } catch (StorageException se) {
-                                                    throw new RuntimeException(
-                                                            "Cannot create PostgresConnection: poolKey = " + poolKey, se);
+                                            IPool pool = new Pool(
+                                                connectionOptions.getMaxConnections(),
+                                                () -> {
+                                                    try {
+                                                        return new PostgresConnection(connectionOptions);
+                                                    } catch (StorageException se) {
+                                                        throw new RuntimeException(
+                                                                "Cannot create PostgresConnection: poolKey = " + poolKey, se);
+                                                    }
+                                                },
+                                                (connection) -> {
+                                                    try {
+                                                        return ((PostgresConnection) connection).validate();
```

Comment:
:thinking: Is this `validate()` function something defined in another lib? How it works?

Reply:
It's defined in another module where PostgresConnection is defined, it does SELECT(1) with some additional checks.
[Here](https://github.com/SmartTools/smartactors-core/blob/35d6cae61144305e80b22db04264c14be8405c3f/CoreFeatures/Database-postgresql/PostgresConnection/src/main/java/info/smart_tools/smartactors/database_postgresql/postgres_connection/PostgresConnection.java#L81) is the function